clientd3d/d3drender_skybox.h

```diff
@@ -14,9 +14,30 @@
 #ifndef _D3DRENDERSKYBOX_H
 #define _D3DRENDERSKYBOX_H
 
-void D3DRenderSkyBoxBegin();
-void D3DRenderSkyBox(room_type* room, Draw3DParams* params, room_contents_node* pRNode, 
-	int& angleHeading, int& anglePitch);
+struct SkyboxRenderParams {
+    LPDIRECT3DVERTEXDECLARATION9 decl1dc;
+    d3d_driver_profile gD3DDriverProfile;
+    d3d_render_pool_new gWorldPool;
+    d3d_render_cache_system gWorldCacheSystem;
+    room_type current_room;
+
+    SkyboxRenderParams(
+        LPDIRECT3DVERTEXDECLARATION9 decl1dc,
+        d3d_driver_profile driver_profile,
+        d3d_render_pool_new pool,
+        d3d_render_cache_system cache_system,
+        room_type& current_room)
+        : decl1dc(decl1dc),
+          gD3DDriverProfile(driver_profile),
+          gWorldPool(pool),
+          gWorldCacheSystem(cache_system),
+          current_room(current_room)
```

Comment:
room_type is a big structure, so this winds up doing a memcpy of the whole thing, which is almost surely not what you want.  How about making current_room a pointer?  Also, always dangerous to name two active variables the same thing.  Rename the parameter.